src/lib/http/http_server.pl

```diff
@@ -132,8 +131,14 @@
     ;   Var = Out  
     ).
 
+header([]) --> [].
+header([Key-Value|Headers]) -->
+    format_("~s: ~s\r\n", [Key, Value]),
+    header(Headers).
+
 write_headers(Stream, Headers) :-
-    forall(member(Key-Value, Headers), format(Stream, "~s: ~s\r\n", [Key, Value])).
+    phrase(header(Headers), Cs),
+    format(Stream, "~s", [Cs]).
```

Comment:
This is a good improvement, thank you a lot!

Once `phrase_to_stream/2` (see #691) becomes available, it can be used to directly write the list of characters to the stream as soon as they become known, without manifesting the list in memory.